aspnetcore/blazor/lifecycle.md

```diff
@@ -0,0 +1,173 @@
+
+If `base.SetParametersAync` isn't invoked, the custom code can interpret the incoming parameters value in any way required. For example, there's no requirement to assign the incoming parameters to the properties on the class.
+
+### After parameters are set
+
+<xref:Microsoft.AspNetCore.Components.ComponentBase.OnParametersSetAsync*> and <xref:Microsoft.AspNetCore.Components.ComponentBase.OnParametersSet*> are called when a component has received parameters from its parent and the values are assigned to properties. These methods are executed after component initialization and each time new parameter values are specified:
+
+```csharp
+protected override async Task OnParametersSetAsync()
+{
+    await ...
+}
+```
+
+> [!NOTE]
+> Asynchronous work when applying parameters and property values must occur during the `OnParametersSetAsync` lifecycle event.
+
+```csharp
+protected override void OnParametersSet()
+{
+    ...
+}
+```
+
+### After component render
+
+<xref:Microsoft.AspNetCore.Components.ComponentBase.OnAfterRenderAsync*> and <xref:Microsoft.AspNetCore.Components.ComponentBase.OnAfterRender*> are called after a component has finished rendering. Element and component references are populated at this point. Use this stage to perform additional initialization steps using the rendered content, such as activating third-party JavaScript libraries that operate on the rendered DOM elements.
+
+The `firstRender` parameter for `OnAfterRenderAsync` and `OnAfterRender`:
+
+* Is set to `true` the first time that the component instance is invoked.
+* Ensures that initialization work is only performed once.
+
+```csharp
+protected override async Task OnAfterRenderAsync(bool firstRender)
+{
+    if (firstRender)
+    {
+        await ...
+    }
+}
+```
+
+> [!NOTE]
+> Asynchronous work immediately after rendering must occur during the `OnAfterRenderAsync` lifecycle event.
+>
+> A component doesn't rerender after the completion of any returned <xref:System.Threading.Tasks.Task>. Infinite render loops are avoided.
+
+```csharp
+protected override void OnAfterRender(bool firstRender)
+{
+    if (firstRender)
+    {
+        ...
+    }
+}
+```
+
+`OnAfterRender` *isn't called when prerendering on the server.*
+
+### Suppress UI refreshing
+
+Override <xref:Microsoft.AspNetCore.Components.ComponentBase.ShouldRender*> to suppress UI refreshing. If the implementation returns `true`, the UI is refreshed:
+
+```csharp
+protected override bool ShouldRender()
+{
+    var renderUI = true;
+
+    return renderUI;
+}
+```
+
+`ShouldRender` is called each time the component is rendered.
+
+Even if `ShouldRender` is overridden, the component is always initially rendered.
+
+## State changes
```

Comment:
I think this section would fit in better with a broader description of how and when components get rendered. Right now this section on StateHasChanged feels a bit out of context.

Reply:
... another option is to move it to the *State Management* topic. It's used in an example over there but not discussed.